Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -126,6 +127,44 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
         cardClient.approveOrder(request: cardRequest, context: context)
     }
 
+    func checkoutWithNativeClient(orderId: String?) async throws -> NativeCheckoutResult {
+        guard let orderId = orderId else {
+
+            return NativeCheckoutResult.error(CoreSDKError(code: 0, domain: "Order Id is null", errorDescription: "Order Id is null"))
```

Comment:
I don't think we'd want merchants to build their own errors using `CoreSDKError`. If we look at some of the other examples in this file we'd likely want to bubble up the errors from the service vs a merchants creating their own.

Reply:
this would be refactored and moved to lib, this is for trying placed all code in demo

Reply:
Is there a follow up ticket for that work? If not, we should just print an error here as using `CoreSDKError` directly is not a behavior we want merchants to follow.

Reply:
yep will be creating a follow-up ticket for this

Reply:
Hey @kgangineni - can you post the ticket number here so we make sure this doesn't get forgotten? I agree with @jaxdesmarais that we want our Demo to mirror how a merchant will integrate with (and interact with) our SDK. 

We don't want them constructing NativeCheckoutResult error types. Actually, they shouldn't even be able to. The constructors for internal error and result types should be inaccessible to merchants. <-- This is also work following up on.

Reply:
@scannillo changes ar in this [PR](https://github.com/paypal/iOS-SDK/pull/69)